internal/luks2/cryptsetup.go

```diff
@@ -255,6 +256,27 @@ func (options *FormatOptions) appendArguments(args []string) []string {
 	return args
 }
 
+var runtimeGOARCH = runtime.GOARCH
+
+// selectCipher will return the cipher to use. This is aes-xts-plain64
+// everywhere except on armhf (32bit arm) hardware where the XTS mode
+// cannot be accelerated by the hardware.
+//
+// Note that this is a simple approach, we could run "cryptsetup
+// benchmark" but that seems over engineered (and easy enough to
+// switch if we need in the future).
+func selectCipherAndKeysize() (string, int) {
+	switch runtimeGOARCH {
+	case "arm":
+		// on 32bit arm CPUs xts cannot be accerlated in HW so
+		// cbc is used
```

Comment:
The underlying Arm32 instruction set doesn't have dedicated instructions for AES at all - in this case we're relying on a feature specific to this SoC which can accelerate AES but IIUC relies on a proprietary kernel blob that doesn't expose an API for doing AES-XTS (so i don't think it's strictly a hardware limitation).

Reply:
Should we / can we / may we switch to xchacha20 or some similar, newer, algorithm? AES implemented in software is usually susceptible to sidechannel data leaks.

Reply:
> Should we / can we / may we switch to xchacha20 or some similar, newer, algorithm? AES implemented in software is usually susceptible to sidechannel data leaks.

Ah, you might be missing a bit of context here - this is for a specific enablement that is armv7 based where there is no accelerated CPU instructions but the SoC does contain a crypto module that's capable of accelerated AES. The issue that this PR addresses is that this hardware module is used by the kernel's crypto code via a proprietary blob supplied by the vendor which exposes high level APIs for performing AES encryption with CBC mode and a few others, but it doesn't support XTS and nor does it offer an API that's at a low enough level to implement this. So this PR enables hardware accelerated crypto for this specific enablement, which is the only way that the device is usable.